conan/tools/premake/premakedeps.py

```diff
@@ -1,10 +1,82 @@
+import itertools 
+import glob
+import re
+
 from conan.internal import check_duplicated_generator
 from conans.model.build_info import CppInfo
 from conans.util.files import save
 
-PREMAKE_FILE = "conandeps.premake.lua"
+# Filename format strings
+PREMAKE_VAR_FILE = "conan_{pkgname}_vars{config}.premake5.lua"
+PREMAKE_CONF_FILE = "conan_{pkgname}{config}.premake5.lua"
+PREMAKE_PKG_FILE = "conan_{pkgname}.premake5.lua"
+PREMAKE_ROOT_FILE = "conandeps.premake5.lua"
+
+# File template format strings
+PREMAKE_TEMPLATE_VAR = """
+conan_includedirs_{pkgname}{config} = {{{deps.includedirs}}}
+conan_libdirs_{pkgname}{config} = {{{deps.libdirs}}}
+conan_bindirs_{pkgname}{config} = {{{deps.bindirs}}}
+conan_libs_{pkgname}{config} = {{{deps.libs}}}
+conan_system_libs_{pkgname}{config} = {{{deps.system_libs}}}
+conan_defines_{pkgname}{config} = {{{deps.defines}}}
+conan_cxxflags_{pkgname}{config} = {{{deps.cxxflags}}}
+conan_cflags_{pkgname}{config} = {{{deps.cflags}}}
+conan_sharedlinkflags_{pkgname}{config} = {{{deps.sharedlinkflags}}}
+conan_exelinkflags_{pkgname}{config} = {{{deps.exelinkflags}}}
+conan_frameworks_{pkgname}{config} = {{{deps.frameworks}}}
+"""
+PREMAKE_TEMPLATE_CONF = """
+include "{premake_varfile}"
+function conan_setup_build_{pkgname}{config}()
+{conf_consume_build}
+end
+function conan_setup_link_{pkgname}{config}()
+{conf_consume_link}
+end
+function conan_setup_{pkgname}{config}()
```

Comment:
No idea if this is even possible, but maybe it would be more idiomatic to have functions like ``conan_setup(pkg_name, config)`` as parameters? and then it will be called ``conan_setup(mypkg, Release)`` or something like that, and that would avoid a lot of generated code?

Reply:
I'm going to take a look at how this is possible in lua. I will respond to this as soon as I have more information.

Reply:
I have found a way to do it like suggested. It required rewriting the way the lua variables are generated and consumed. It also made the python code around 50 lines shorter. 

 Doing it like suggested is also removing the need for the `_premake_filtered_fallback` feature. This can now be replaced by a call to `conan_setup(conf)` over the automatic `conan_setup()`.

The lua function calls work in cascades:
- `conan_setup()` will call all possible
- `conan_setup(conf)` that call 
- `conan_setup(conf, pkg)` for all packages

I will do a few more tests with my own codebase to see if the implementation holds up and then do the commit.